### PR TITLE
Sync 20230512 algorand

### PR DESCRIPTION
**Pull request by guplersaxanoid**
# Description
Sync with main sdk

### CODE REVIEW

File: .github/workflows/gh-release.yml [hunk 1]
```diff
@@ -25,6 +25,12 @@ jobs:
           package-path: packages/common-algorand
           repo-token: ${{ secrets.REPO_TOKEN }}
 
+
+      - uses: ./.github/actions/create-release
```
Comment (stwiname) [inline .github/workflows/gh-release.yml]: Can be removed

File: .github/workflows/prerelease.yml [hunk 2]
```diff
@@ -45,6 +45,11 @@ jobs:
         with:
           paths: packages/common-algorand
 
+      - uses: marceloprado/has-changed-path@v1
```
Comment (stwiname) [inline .github/workflows/prerelease.yml]: Can be removed

File: .github/workflows/prerelease.yml [hunk 3]
```diff
@@ -71,6 +76,20 @@ jobs:
...
+        with:
+          package-path: packages/testing
+          npm-token: ${{ secrets.NPM_TOKEN }}
+
+      - name: Bump node-core & deploy
+        if: steps.changed-node-core.outputs.changed == 'true'
+        uses: ./.github/actions/create-prerelease
+        with:
+          package-path: packages/node-core
+          npm-token: ${{ secrets.NPM_TOKEN }}
```
Comment (stwiname) [inline .github/workflows/prerelease.yml]: Same here

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -46,6 +46,11 @@ jobs:
         with:
           paths: packages/common-algorand
 
+      - uses: marceloprado/has-changed-path@v1
```
Comment (stwiname) [inline .github/workflows/release.yml]: Same in this file

File: packages/common-algorand/package.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
   "main": "dist/index.js",
   "license": "Apache-2.0",
   "dependencies": {
-    "@subql/common": "^2.0.0",
+    "@subql/common": "^2.1.1",
     "@subql/types-algorand": "workspace:*",
     "class-transformer": "0.4.0",
     "class-validator": "^0.13.2",
```
Comment (stwiname) [inline packages/common-algorand/package.json:19]: Can be removed

File: packages/common-algorand/test/project_1.0.0_node_options.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
+specVersion: 1.0.0
```
Comment (stwiname) [inline packages/common-algorand/test/project_1.0.0_node_options.yaml]: This isn't for algorand

File: packages/node/src/indexer/dictionary.service.ts [hunk 7]
```diff
@@ -4,6 +4,8 @@
 import { Inject, Injectable, OnApplicationShutdown } from '@nestjs/common';
 import {
   NodeConfig,
+  timeout,
+  getLogger,
```
Comment (stwiname) [inline packages/node/src/indexer/dictionary.service.ts]: Unused?

File: packages/node/src/indexer/fetch.service.ts [hunk 8]
```diff
@@ -158,99 +142,27 @@ export class FetchService implements OnApplicationShutdown {
...
-    return this.useDictionary;
+  protected async getFinalizedHeight(): Promise<number> {
+    const checkHealth = await this.api.api.makeHealthCheck().do();
+    return checkHealth.round;
   }
 
-  getLatestFinalizedHeight(): number {
-    return this.latestFinalizedHeight;
+  protected async getBestHeight(): Promise<number> {
+    return this.getFinalizedHeight();
```
Comment (stwiname) [inline packages/node/src/indexer/fetch.service.ts:151]: Does algorand have instant finalization?

File: packages/node/src/indexer/unfinalizedBlocks.service.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  protected async getFinalizedHead(): Promise<Header> {
+    const checkHealth = await this.apiService.api.api.makeHealthCheck().do();
+    const latestHeight = checkHealth.round;
+    return this.getHeaderForHeight(latestHeight);
+  }
+
+  protected async getHeaderForHash(hash: string): Promise<Header> {
+    return this.getHeaderForHeight(parseInt(hash, 10));
```
Comment (stwiname) [inline packages/node/src/indexer/unfinalizedBlocks.service.ts:43]: Is this right? Shouldn't the hash be a string?

File: packages/node/test/projectFixture/template-v1.0.0/project.yaml [hunk 10]
```diff
@@ -0,0 +1,91 @@
+specVersion: 1.0.0
```
Comment (stwiname) [inline packages/node/test/projectFixture/template-v1.0.0/project.yaml]: Doesn't relate to algorand, same with the whole directory

File: packages/node/src/indexer/fetch.service.ts (hunk 8 above)
Comment (guplersaxanoid) [reply packages/node/src/indexer/fetch.service.ts:151]: yes

**Review by stwiname — APPROVED** [review]

**Pull request merged**